src/frontend/screens/Sync/index.tsx

```diff
@@ -0,0 +1,211 @@
+
+  React.useEffect(() => {
+    const syncListener = (val: State) => {
+      setConnectedPeers(val.connectedPeers);
+      setIsSyncing(val.data.syncing);
+      setHasDataToSync(val.data.dataToSync);
+    };
+
+    sync.addListener('sync-state', syncListener);
+
+    return () => {
+      sync.removeListener('sync-state', syncListener);
+    };
+  }, [sync]);
+
+  React.useLayoutEffect(() => {
+    if (ssid)
+      navigation.setOptions({
+        headerTitle: () => (
+          <View style={{flexDirection: 'row', alignItems: 'center'}}>
+            <MaterialIcons name="wifi" size={20} color={BLACK} />
+            <Text style={{marginLeft: 5}}>{ssid}</Text>
+          </View>
+        ),
+        headerRight: () => <SettingsButton />,
+      });
+  }, [ssid]);
+
+  React.useLayoutEffect(() => {
+    if (!ssid) {
+      navigation.navigate('NoWifi');
+    }
+  }, [ssid, navigation]);
+
+  return (
+    <View style={styles.container}>
+      <View>
+        {projectSettings.data && projectSettings.data.name && (
+          <Text style={{textAlign: 'center'}}>{projectSettings.data.name}</Text>
+        )}
+        {deviceInfo.data && deviceInfo.data.name && (
+          <Text style={{textAlign: 'center'}}>
+            {formatMessage(m.yourName, {deviceName: deviceInfo.data.name})}
+          </Text>
+        )}
+        {connectedPeers !== undefined && (
+          <View style={styles.flexRow}>
+            <MaterialIcons name="wifi" size={20} color={BLACK} />
+            <Text style={{marginLeft: 10}}>
+              {formatMessage(m.connectedPeers, {
+                numberOfDevices: connectedPeers,
+              })}
+            </Text>
+          </View>
+        )}
+        {isSyncing ? (
+          <View>{/* syncing here */}</View>
+        ) : (
+          <Text style={{textAlign: 'center'}}>devices want to sync</Text>
+        )}
+      </View>
+
+      <SyncButton
+        startSync={sync.start}
+        stopSync={sync.stop}
+        isSyncing={false}
+        hasDataToSync={true}
+      />
+    </View>
+  );
+};
+
+const SyncButton = ({
+  isSyncing,
+  hasDataToSync,
+  stopSync,
+  startSync,
+}: {
+  isSyncing: boolean;
+  hasDataToSync?: boolean;
+  stopSync: () => void;
+  startSync: () => void;
+}) => {
+  const {formatMessage} = useIntl();
+  if (isSyncing) {
+    return (
+      <Button variant="outlined" fullWidth onPress={stopSync}>
+        <View style={styles.flexRow}>
+          <MaterialIcons name="stop" color={BLACK} size={40} />
+          <Text style={{fontSize: 16, marginLeft: 5}}>
+            {formatMessage(m.stop)}
+          </Text>
+        </View>
+      </Button>
+    );
+  }
+
+  if (hasDataToSync === undefined) {
+    return null;
+  }
+
+  if (!hasDataToSync) {
+    return (
+      <Button fullWidth onPress={() => {}}>
+        {formatMessage(m.caughtUp)}
+      </Button>
+    );
+  }
+
+  return (
+    <Button fullWidth onPress={startSync}>
```

Comment:
Should the user be able to press [sync.start()](https://github.com/digidem/mapeo-core-docs/tree/main/client-docs#syncstart) even if there are no connected peers

Reply:
A question for the co-design team I think. From a technical perspective this makes sense: it puts the peer in "syncing mode" where any peers that join will be able to start syncing.